x-pack/plugins/cases/public/components/edit_connector/index.tsx

```diff
@@ -133,79 +104,44 @@ export const EditConnector = React.memo(
     const { actions } = useApplicationCapabilities();
     const actionsReadCapabilities = actions.read;
 
-    // by default save if disabled
-    const [enableSave, setEnableSave] = useState(false);
-
     const { setFieldValue, submit } = form;
 
     const [{ currentConnector, fields, editConnector }, dispatch] = useReducer(
       editConnectorReducer,
-      { ...initialState, fields: caseFields }
+      {
+        ...initialState,
+        fields: caseFields,
+        currentConnector: getConnectorById(caseData.connector.id, allAvailableConnectors),
+      }
     );
 
     // only enable the save button if changes were made to the previous selected
     // connector or its fields
-    useEffect(() => {
-      // null and none are equivalent to `no connector`.
-      // This makes sure we don't enable the button when the "no connector" option is selected
-      // by default. e.g. when a case is created without a selector
-      const isNoConnectorDeafultValue =
-        currentConnector === null && selectedConnector === NONE_CONNECTOR_ID;
-      const enable =
-        (!isNoConnectorDeafultValue && currentConnector?.id !== selectedConnector) ||
-        !deepEqual(fields, caseFields);
+    // null and none are equivalent to `no connector`.
+    // This makes sure we don't enable the button when the "no connector" option is selected
+    // by default. e.g. when a case is created without a selector
+    const isNoConnectorDefaultValue =
+      currentConnector === null && selectedConnector === NONE_CONNECTOR_ID;
 
-      setEnableSave(enable);
-    }, [caseFields, currentConnector, fields, selectedConnector]);
-
-    useEffect(() => {
```

Comment:
This hook used to update the current connector when `caseData.connector` changes. I accomplished the same with using the `key` property of the `EditConnector` component. Take a look at https://github.com/elastic/kibana/pull/149276/files?diff=split&w=1#r1084004033 and https://github.com/elastic/kibana/pull/149276/files?diff=split&w=1#r1083999611